Style Guide

lhirata opened the pull request:
We strive for clear, correct, and concise documentation! Here's our style guide.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/docs/3032)
<!-- Reviewable:end -->

lhirata commented:
@jseldess @Amruta-Ranade @rmloveland - here's a draft of the style guide. It's about 80% of the way to being reviewable. If you want to start a casual review, feel free. I mostly want to make sure I'm on track, but there is some cleanup / additional information I still need to write.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/e4e02e48cbff372ef8d7ba09966369d9e5901395/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/ee3eeceb4b7be799e4ddea58a52a4b5ccaff6bf7/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/3931717e27730ffd26ab16a828019923938f9d59/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/9aa77e7bea95eb5a46f32f10788139dbf8fae469/

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+
+### Headings
+
+Use headings to demarcate content into a hierarchy to help readers find information easier.
```
Suggestion: We can also include "Don't use colons after headings". I keep repeating this mistake, so it will help me remember :)

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+
+Use images to clarify a topic, and should be used only as needed. Images are either:
+
+- **Screenshots** - Provide a UI visual. Screenshots should show enough of the UI that the user can easily orient themselves and understand what they are being shown. If a screenshot needs an annotation, use a red box.
```
Red box? Not sure I have seen one in our docs (should I use red boxes in Admin UI docs? 🤔)

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+Use the following HTML and liquid to include an image in a markdown page:
+
+~~~ html
+<img src="{% raw %}{{ 'images/v2.1/image-name.png' | relative_url }}{% endraw %}" alt="Alternative Text Here" style="border:1px solid #eee;max-width:100%" />
```
We need to use {% raw %}? God, Lauren, I am so glad you took up this project. I am learning so much!!

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+
+on the line directly preceding the code block, for example:
+
+<pre>
```
What's pre?

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+
+when to move to html?
+
+We do not use outer pipes.
```
Again, so good to know!

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+~~~
+</pre>
+
+**Copy to Clipboard** should be used for every code block that can be executed.
```
Suggestion: Put "executed" in bold. That's important info, especially for external contributors.

Amruta-Ranade commented on style-guide.md:
```diff
@@ -0,0 +1,311 @@
+
+When?
+
+### Version Tags
```
Can you also provide guidance on where to put these tags? I spend a lot of time figuring out the correct placement in headings, sections, in a table, in examples, etc.

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/72601022a39b32e241cccb3c98cdfbd0045aa2c2/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/75fad2367388e6e7635613b28dcac28c4d8fec6f/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/73f3694ddc16b6a0294e7225a804a21d452daec9/

lhirata commented:
Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[style-guide.md, line 24 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-24:-LBXkiun84FdjS1kL64X:bgumddr) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L24)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

"Grammar can be incorrect to save simplicity." 
This sounds like we are okay with incorrect grammar? 
Suggestion: I think we can delete this sentence.
</blockquote></details>

I added an example of an instance where the grammar is incorrect but is a lot more simple than the correct way (where you don't end with a preposition). Do you think this is still confusing / unnecessary to call out?

---

*[style-guide.md, line 85 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-85:-LBXlAHY7EkyR3_nkRFg:b98geu2) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L85)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

What's pre?
</blockquote></details>

Ignore this!

---

*[style-guide.md, line 92 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-92:-LBXlNqy2UPQUalZK4wV:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L92)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Suggestion: Put "executed" in bold. That's important info, especially for external contributors.
</blockquote></details>

Done.

---

*[style-guide.md, line 103 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-103:-LBXlOx2DoZ3ZOstTNyc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L103)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Suggestion: We can also include "Don't use colons after headings". I keep repeating this mistake, so it will help me remember :)
</blockquote></details>

Done.

---

*[style-guide.md, line 116 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-116:-LBXlPtN5s66lLWDuFNt:b5fw5tc) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L116)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Red box? Not sure I have seen one in our docs (should I use red boxes in Admin UI docs? 🤔)
</blockquote></details>

Added an example below where a red box is used. The idea here is that if you _need_ to annotate a screenshot, we should dictate some kind of standard.

---

*[style-guide.md, line 122 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-122:-LBXlcIt21BaE4yV8rJ4:bl0nxze) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L122)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

We need to use {% raw %}? God, Lauren, I am so glad you took up this project. I am learning so much!!
</blockquote></details>

Ignore!

---

*[style-guide.md, line 303 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-303:-LBXmC6hBEirlz8v-jb9:bxhlng9) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L303)):*
<details><summary><i>Previously, Amruta-Ranade (Amruta Ranade) wrote…</i></summary><blockquote>

Can you also provide guidance on where to put these tags? I spend a lot of time figuring out the correct placement in headings, sections, in a table, in examples, etc.
</blockquote></details>

Added a little bit of guidance but I think we might need to extrapolate more. Thoughts?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/

jseldess commented:
This is so excellent, @lhirata! Left some comments and questions. 

One gap I didn't mention anywhere is that we don't talk about the "Concept" doc type, which right now is pretty much just the architecture docs. What should we do about that?

---

Review status: 0 of 2 files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[style-guide.md, line 24 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-24:-LBa_obK9hM1mfD8C_5t:bjdd2kh) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L24)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

I added an example of an instance where the grammar is incorrect but is a lot more simple than the correct way (where you don't end with a preposition). Do you think this is still confusing / unnecessary to call out?
</blockquote></details>

I think this is an important rule (that it's ok to use phrases instead of complete sentences in some cases), but I wonder if we should mention that this is mostly relevant for definitions or descriptions in reference docs?

Btw, I think this style guide is an excellent demonstration of the guidance to "use simple and direct language"!!!!

---

*[style-guide.md, line 116 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-116:-LBameB89zgA3n6DESdJ:ba3udv1) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L116)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Added an example below where a red box is used. The idea here is that if you _need_ to annotate a screenshot, we should dictate some kind of standard.
</blockquote></details>

I'm actually not sure if I agree that screenshots should show "enough of the UI that the user can easily orient themselves". The UI changes enough that this approach will mean a lot of image rot. We may want to limit screenshots to the specific area in focus when it's clear where to go to see that. Amruta updated lots of images in this way for the 2.0 release, so she may have more thoughts.

---

*[STYLE.md, line 13 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaZXrF0J4fu6Z-nWnw:-LBaZXrF0J4fu6Z-nWnx:b-92expu) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L13)):*
> ```Markdown
> **Note:** This style guide should be viewed in its [rendered state](https://github.com/cockroachdb/docs/blob/master/STYLE.md), not in raw Markdown.
> 
> <div id="toc"></div>
> ```

This isn't going to work cause we're not running this through Jekyll. To create a page toc, you have to do it manually. For example, ee the cockroach [`CONTRIBUTING.md`](https://raw.githubusercontent.com/cockroachdb/cockroach/master/README.md) file.

---

*[STYLE.md, line 36 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaaUDTBfxE4ue3w0rw:-LBaaUDU2O8GSPVZlgUM:b-uns385) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L36)):*
> ```Markdown
>     **Example:** `table name`: The name of the table you want to create audit logs for.
> 
> - Use contractions to simplify language.
> ```

Provide an example or two?

---

*[STYLE.md, line 42 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaalWa-q03lFMKOboY:-LBaalWa-q03lFMKOboZ:b-r98nw0) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L42)):*
This is obvious to us, I think, but an example might help someone who isn't clear on what "title case" means exactly.  We might also contrast it to "sentence case", e.g.:

> Use title case instead of sentence case for all headings.

---

*[STYLE.md, line 43 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBab2AN0DkPIIlQjJk1:-LBab2AN0DkPIIlQjJk2:box9ai) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L43)):*
> ```Markdown
> 
> - Use title case for all [headings](#headings).
> - Depending on the context, use title case or sentence case for [links](#links).
> ```

Having gotten there yet, but I'm assuming examples in the links section clarify this further?

---

*[STYLE.md, line 52 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBabBZTDETsWaQhIVQU:-LBabBZTDETsWaQhIVQV:b-bj22y7) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L52)):*
> ```Markdown
> Punctuation rules:
> 
> - Use periods instead of semicolons.
> ```

No room for semicolons at all? 

---

*[STYLE.md, line 58 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBacLLTEXeu8Pm4DPxn:-LBacLLTEXeu8Pm4DPxo:b5aefhu) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L58)):*
> ```Markdown
> ## File Conventions
> 
> CockroachDB docs are mainly comprised of pages (`.md`) and images (`.png` or `.gif`). File names are lowercase with a dash between words, and should be brief but descriptive.
> ```

This is a solid convention for now. Eventually, we may want to get more prescriptive about image naming in particular; it's often very hard to know where an image is used based on the name, though we can figure that out via a quick search in a text editor, of course. 

---

*[STYLE.md, line 73 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBacs6b26UUoITUtSHF:-LBacs6b26UUoITUtSHG:bk2pkva) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L73)):*
> ```Markdown
> ### Tutorials, Training, and Examples
> 
> Tutorials, training, and examples are educational docs or sections, meant to acquaint users with CockroachDB and its features. These docs should be written in a more conversational [tone](#language-and-tone), as if they are teaching the user.
> ```

nit: remove comma before `meant to`.

---

*[STYLE.md, line 75 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaclJwDiF47s2Fe0-S:-LBaclJwDiF47s2Fe0-T:bf7o7bh) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L75)):*
> ```Markdown
> Tutorials, training, and examples are educational docs or sections, meant to acquaint users with CockroachDB and its features. These docs should be written in a more conversational [tone](#language-and-tone), as if they are teaching the user.
> 
> Examples:
> ```

Can you clarify which types these examples represent? 

---

*[STYLE.md, line 84 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBadP39EUJ696qJ1o7G:-LBadP39EUJ696qJ1o7H:b9fbtxu) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L84)):*
I'd suggest a slight rephrasing:

> Reference docs are informational and provide a comprehensive description of a CockroachDB function or feature, while task-based docs are instructional and provide prescriptive, step-by-step guidance to help a user complete a specific task.

---

*[STYLE.md, line 86 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaeMj72Y7gu6hq2JpX:-LBaeMj72Y7gu6hq2JpY:b-ze6ag3) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L86)):*
> ```Markdown
> Reference and task-based docs are informational docs that provide details about a CockroachDB function or feature, as well as step-by-step instructions to help the user complete a specific task.
> 
> Examples:
> ```

Again, I think we may want to explicitly call out the types here. I'd suggest we show SQL Reference, CLI reference, and task. For task, perhaps use something like https://www.cockroachlabs.com/docs/stable/orchestrate-cockroachdb-with-kubernetes.html and/or https://www.cockroachlabs.com/docs/stable/upgrade-cockroach-version.html.

---

*[STYLE.md, line 139 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBalcblDE5sVwwynQev:-LBalcblDE5sVwwynQew:b-f1gi0p) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L139)):*
> ```Markdown
> Examples help show the feature in action. Examples follow a basic format:
> 
> 1. The **Title** should start with a verb and should describe the task the example is outlining. It should use title-case.
> ```

Is it `title case` or `title-case`? Think we need to make sure we're consistent, whichever we use.

---

*[STYLE.md, line 143 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBallJVAvX_g-ZufEyX:-LBallJVAvX_g-ZufEyY:bfzr9mq) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L143)):*
> ```Markdown
>     **Example:** Create a Table that Mirrors Key-Value Storage
> 
> 2. **Introductory information** should be provided if some context is need to orient the user and can also be used to introduce code blocks. This should be written in a conversational tone.
> ```

`is need` > `is needed`

---

*[STYLE.md, line 161 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBam4GsAhmiPZLFW--7:-LBam4GsAhmiPZLFW--8:b-80laid) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L161)):*
> ```Markdown
> Use headings to demarcate content into a hierarchy to help readers find information easier.
> 
> Headings should be title case, and are denoted by octothorps (`#`) followed by one space. Enter a line break between a heading and its content. CockroachDB docs use Heading 2 (`##`) and Heading 3 (`###`). Heading 1 is reserved for page titles and anything under Heading 3 can be denoted by bolded text, or should be reformatted.
> ```

Whoa, I learned a word: `octothorps`!

---

*[STYLE.md, line 170 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBamPPZBY94IsO7HCto:-LBamPPZBY94IsO7HCtp:b-6krf9a) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L170)):*
> ```Markdown
> ### Images
> 
> Use images to clarify a topic, and should be used only as needed. Images are either:
> ```

Something is off about the second part of the first sentence.

---

*[STYLE.md, line 192 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBanShV9Vj8Sbyjha4t:-LBanShV9Vj8Sbyjha4u:bqzli51) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L192)):*
Might want to clarify the second sentence, e.g.:

> - For an internal link to another page in the docs, use just the name of the file, for example, `[here](name-of-page.html)`.
> - For an internal link to a specific section on another page, use the name of the file plus the heading, for example, `[xyz](name-of-page.html#heading-on-page)`.
> - For an internal link to a section on the current page, use just the heading, for example, `[xyz](#heading-on-page)`.  

---

*[STYLE.md, line 209 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaoqG57FjaFj40b2sv:-LBaoqG57FjaFj40b2sw:bva2bch) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L209)):*
> ```Markdown
> - **Bulleted** (i.e., unordered list) - Use to list related information in an easy-to-read way.
> 
>     **Example:** This list is a bulleted list.
> ```

Did you mean to use something else as an example?

---

*[STYLE.md, line 221 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBapERe0AYukN_ArPX7:-LBap5IXCT2Nc98UbBRW:b4zm96n) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L221)):*
> ```Markdown
> To nest a list under a list item, start the list on the next line (no empty line), and indent the new list four spaces, for example:
> 
> ~~~
> ```

I think you might need to use three backticks instead of the three ~s here and in a few other places.

---

*[STYLE.md, line 330 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBapXcs9Y5BvZ1QqL4h:-LBapXcs9Y5BvZ1QqL4i:b-l6tnf4) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L330)):*
> ```Markdown
> ~~~
> 
> **Note:**  If a table becomes too complex, format it in HTML.
> ```

I'm unsure if this is guidance we want to provide, given that we don't have existing examples. 

Also, we probably need to list some of the limitations of markdown tables, for example, that you can use basic markdown formatting (bold, italics, inline code, links) but need to use html for things like paragraph breaks and bullets. Also, I don't think you can include into a table cell or use code block. We might need to dig in a bit to figure this all out. 

---

*[STYLE.md, line 332 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBac5HW0Xj-X_NKY4mI:-LBac5HW0Xj-X_NKY4mJ:b-dk5ryw) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L332)):*
> ```Markdown
> **Note:**  If a table becomes too complex, format it in HTML.
> 
> ### Text Format
> ```

Feels like this should be much further up, perhaps following `Capitalization and Punctuation`.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032#-:-LBako1g02yA2sbn4YOV:b4vrv3b)*
<!-- Sent from Reviewable.io -->

lhirata commented:
Review status: 0 of 2 files reviewed at latest revision, 27 unresolved discussions, all commit checks successful.

---

*[style-guide.md, line 24 at r1](https://reviewable.io/reviews/cockroachdb/docs/3032#-LB2jgNJuQAZRCOfd4SU-r1-24:-LBb6HNID2ru9E4IybcD:b-eowr4b) ([raw file](https://github.com/cockroachdb/docs/blob/9aa77e7bea95eb5a46f32f10788139dbf8fae469/style-guide.md#L24)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think this is an important rule (that it's ok to use phrases instead of complete sentences in some cases), but I wonder if we should mention that this is mostly relevant for definitions or descriptions in reference docs?

Btw, I think this style guide is an excellent demonstration of the guidance to "use simple and direct language"!!!!
</blockquote></details>

Clarified!

---

*[STYLE.md, line 13 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaZXrF0J4fu6Z-nWnw:-LBb4xrd0CXPB8I7a0M3:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L13)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This isn't going to work cause we're not running this through Jekyll. To create a page toc, you have to do it manually. For example, ee the cockroach [`CONTRIBUTING.md`](https://raw.githubusercontent.com/cockroachdb/cockroach/master/README.md) file.
</blockquote></details>

Done.

---

*[STYLE.md, line 36 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaaUDTBfxE4ue3w0rw:-LBb7bEN4AgK638pCHI_:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L36)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Provide an example or two?
</blockquote></details>

Done.

---

*[STYLE.md, line 42 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaalWa-q03lFMKOboY:-LBb7tdHB24hNmVohOvj:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L42)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

This is obvious to us, I think, but an example might help someone who isn't clear on what "title case" means exactly.  We might also contrast it to "sentence case", e.g.:

> Use title case instead of sentence case for all headings.
</blockquote></details>

Done.

---

*[STYLE.md, line 43 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBab2AN0DkPIIlQjJk1:-LBb7wJgA_jq5vt-8ybL:b-8jvxzz) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L43)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Having gotten there yet, but I'm assuming examples in the links section clarify this further?
</blockquote></details>

Yeah, the Links section describes the scenarios 

---

*[STYLE.md, line 52 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBabBZTDETsWaQhIVQU:-LBb90kV0FhHdB9UKgYQ:b-jfu3cc) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L52)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

No room for semicolons at all? 
</blockquote></details>

Adjusted the language

---

*[STYLE.md, line 73 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBacs6b26UUoITUtSHF:-LBb8V094JGtkFVmrN9h:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L73)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

nit: remove comma before `meant to`.
</blockquote></details>

Done.

---

*[STYLE.md, line 75 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaclJwDiF47s2Fe0-S:-LBb9TmCDxu0yp0Cd-o_:bhuu6lo) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L75)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Can you clarify which types these examples represent? 
</blockquote></details>

Clarified

---

*[STYLE.md, line 84 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBadP39EUJ696qJ1o7G:-LBb9aZj79xVq5AU_14b:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L84)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'd suggest a slight rephrasing:

> Reference docs are informational and provide a comprehensive description of a CockroachDB function or feature, while task-based docs are instructional and provide prescriptive, step-by-step guidance to help a user complete a specific task.
</blockquote></details>

Done.

---

*[STYLE.md, line 86 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaeMj72Y7gu6hq2JpX:-LBbB2vp0L7q7NFE-37X:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L86)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Again, I think we may want to explicitly call out the types here. I'd suggest we show SQL Reference, CLI reference, and task. For task, perhaps use something like https://www.cockroachlabs.com/docs/stable/orchestrate-cockroachdb-with-kubernetes.html and/or https://www.cockroachlabs.com/docs/stable/upgrade-cockroach-version.html.
</blockquote></details>

Done.

---

*[STYLE.md, line 139 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBalcblDE5sVwwynQev:-LBb8JXg2ge718rFG7LO:b-75gbrz) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L139)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Is it `title case` or `title-case`? Think we need to make sure we're consistent, whichever we use.
</blockquote></details>

Edited so it's title-case and sentence-case.

---

*[STYLE.md, line 143 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBallJVAvX_g-ZufEyX:-LBbBAK66SC9T_DG9Euc:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L143)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`is need` > `is needed`
</blockquote></details>

Done.

---

*[STYLE.md, line 161 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBam4GsAhmiPZLFW--7:-LBbBBO50ZUktoyve9-i:bymrri5) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L161)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Whoa, I learned a word: `octothorps`!
</blockquote></details>

Ha! If it's too pedantic, I can switch this to "hash" or "pound sign".

---

*[STYLE.md, line 170 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBamPPZBY94IsO7HCto:-LBbBl1a1i_pMudZi9g5:bt2vr1b) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L170)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Something is off about the second part of the first sentence.
</blockquote></details>

Fixed.

---

*[STYLE.md, line 192 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBanShV9Vj8Sbyjha4t:-LBbHIzi38iOU9epGqQH:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L192)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Might want to clarify the second sentence, e.g.:

> - For an internal link to another page in the docs, use just the name of the file, for example, `[here](name-of-page.html)`.
> - For an internal link to a specific section on another page, use the name of the file plus the heading, for example, `[xyz](name-of-page.html#heading-on-page)`.
> - For an internal link to a section on the current page, use just the heading, for example, `[xyz](#heading-on-page)`.  
</blockquote></details>

Done.

---

*[STYLE.md, line 209 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaoqG57FjaFj40b2sv:-LBbHNbt8QMzx_Sb-Jlh:b-br471y) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L209)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Did you mean to use something else as an example?
</blockquote></details>

No, I meant that the list you're reading is a bulleted list :) 

I changed the example to be clearer.

---

*[STYLE.md, line 221 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBapERe0AYukN_ArPX7:-LBbLHQo12JwSJYdBj-b:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L221)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I think you might need to use three backticks instead of the three ~s here and in a few other places.
</blockquote></details>

Done.

---

*[STYLE.md, line 330 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBapXcs9Y5BvZ1QqL4h:-LBbLKpfETWHpCq5xsxQ:bulhuyn) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L330)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

I'm unsure if this is guidance we want to provide, given that we don't have existing examples. 

Also, we probably need to list some of the limitations of markdown tables, for example, that you can use basic markdown formatting (bold, italics, inline code, links) but need to use html for things like paragraph breaks and bullets. Also, I don't think you can include into a table cell or use code block. We might need to dig in a bit to figure this all out. 
</blockquote></details>

One of Rich's docs includes a HTML table, but I'll remove this for now since it doesn't seem to be a common issue (yet).

Added more details about the limitations

---

*[STYLE.md, line 332 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBac5HW0Xj-X_NKY4mI:-LBbNbL40P504ZLyZqEL:b-4hpj9m) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L332)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Feels like this should be much further up, perhaps following `Capitalization and Punctuation`.
</blockquote></details>

It's low on the page because I organized the sections alphabetically. Do you think it would be better to arrange by whatever's used the most?

(also, I added a link from Capitalization and Punctuation to Text Format)

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/d1969c44ba126bf63e8a398beacac2680935c6dd/

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/0f86801a2b4f55e5562788a6970571864605f2ec/

jseldess commented:
*[STYLE.md, line 430 at r5](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBcpJUL8KelDWK7t3af:-LBcpJUL8KelDWK7t3ag:b-wkw83t) ([raw file](https://github.com/cockroachdb/docs/blob/0f86801a2b4f55e5562788a6970571864605f2ec/STYLE.md#L430)):*
> ```Markdown
> Examples:
> 
> - [`CREATE TABLE`](create-table.html)
> ```

This link, and other such links to pages/files in our docs, are broken. If you want to link to a page/file in our docs, you need to use the full url, since this style page isn't run through jekyll and part of our site. 

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[STYLE.md, line 112 at r5](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBcpzq31ekyVfc0i3_e:-LBcpzq31ekyVfc0i3_f:b9n6srv) ([raw file](https://github.com/cockroachdb/docs/blob/0f86801a2b4f55e5562788a6970571864605f2ec/STYLE.md#L112)):*
> ```Markdown
> 
> - **SQL reference doc example:** [`CREATE TABLE`](create-table.html)
> - **CLI reference doc example:** [Start a Local Cluster](start-a-local-cluster.html)
> ```

This isn't a cli reference example. For that, use https://www.cockroachlabs.com/docs/stable/use-the-built-in-sql-client.html. 

Also, as mentioned further down, these links are broken. If you want to link to the live versions of these pages, you need to include the full urls.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

jseldess commented:
*[STYLE.md, line 161 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBam4GsAhmiPZLFW--7:-LBcqOz78nltmWX6rRJq:bwu8xwm) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L161)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

Ha! If it's too pedantic, I can switch this to "hash" or "pound sign".
</blockquote></details>

Yeah, I think many more people are likely to understand either of those two alternative terms.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

jseldess commented:
Review status: 0 of 2 files reviewed at latest revision, 17 unresolved discussions, all commit checks successful.

---

*[STYLE.md, line 209 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBaoqG57FjaFj40b2sv:-LBcqiGRCzYblyMfs7pX:b-rcuav3) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L209)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

No, I meant that the list you're reading is a bulleted list :) 

I changed the example to be clearer.
</blockquote></details>

So on those linked pages, we should see examples of a numbered and bulleted list? If you fix the links, I'll validate.

---

*[STYLE.md, line 332 at r3](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBac5HW0Xj-X_NKY4mI:-LBcr4V8BmJpGCQnID-E:b-lggnox) ([raw file](https://github.com/cockroachdb/docs/blob/0f880fe28f0d25c190ecc9965bfb612d66e4ed4a/STYLE.md#L332)):*
<details><summary><i>Previously, lhirata wrote…</i></summary><blockquote>

It's low on the page because I organized the sections alphabetically. Do you think it would be better to arrange by whatever's used the most?

(also, I added a link from Capitalization and Punctuation to Text Format)
</blockquote></details>

Ah, I see. Yeah, I think it'd be better to order in terms of relevance for our daily work. Same for the doc types, for which Concepts would probably come last.

---

*[STYLE.md, line 96 at r5](https://reviewable.io/reviews/cockroachdb/docs/3032#-LBcpvSP1sKgB17Hbae_:-LBcpvSP1sKgB17Hbaea:b3ksk) ([raw file](https://github.com/cockroachdb/docs/blob/0f86801a2b4f55e5562788a6970571864605f2ec/STYLE.md#L96)):*
> ```Markdown
> ## Documentation Types
> 
> ### Concept Docs
> ```

i

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/docs/3032)*
<!-- Sent from Reviewable.io -->

Amruta-Ranade commented:
This is incredible! LGTM (is an understatement) 🎉

cockroach-teamcity commented:
http://cockroach-docs-review.s3-website-us-east-1.amazonaws.com/c9a946d6f855300c11357dbb628a764a52c8aeb4/

jseldess commented:
@lhirata, I'm merging this so we have an updated style guide for WTD. You can make follow-up edits in a separate PR and ask @rmloveland and others to be reviewers on that as well.
